akka-docs/src/test/java/jdocs/typed/tutorial_3/inprogress2/Device.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright (C) 2018 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package jdocs.typed.tutorial_3.inprogress2;
+
+//#device-with-read
+
+import akka.actor.typed.Behavior;
+import akka.actor.typed.javadsl.AbstractBehavior;
+import akka.actor.typed.javadsl.ActorContext;
+import akka.actor.typed.javadsl.Behaviors;
+import akka.actor.typed.javadsl.Receive;
+
+import java.util.Optional;
+
+//#device-with-read
+import static jdocs.typed.tutorial_3.inprogress2.DeviceProtocol.*;
+/*
+//#device-with-read
+import static com.lightbend.akka.sample.DeviceProtocol.*;
+//#device-with-read
+*/
+//#device-with-read
+
+public class Device extends AbstractBehavior<DeviceMessage> {
+
+  public static Behavior<DeviceMessage> behavior(String groupId, String deviceId) {
+    return Behaviors.setup(context -> new Device(context, groupId, deviceId));
+  }
+
+  private final ActorContext<DeviceMessage> context;
+  private final String groupId;
+  private final String deviceId;
+
+  private Optional<Double> lastTemperatureReading = Optional.empty();
+
+  public Device(ActorContext<DeviceMessage> context, String groupId, String deviceId) {
+    this.context = context;
+    this.groupId = groupId;
+    this.deviceId = deviceId;
+
+    context.getLog().info("Device actor {}-{} started", groupId, deviceId);
+  }
+
+  @Override
+  public Receive<DeviceMessage> createReceive() {
+    return receiveBuilder()
+      .onMessage(ReadTemperature.class, this::readTemperature)
+      .build();
+  }
+
+  private Behavior<DeviceMessage> readTemperature(ReadTemperature r) {
+    r.replyTo.tell(new RespondTemperature(r.requestId, lastTemperatureReading));
+    return this;
+  }
+
+  private Device postStop() {
```

Comment:
needs to be updated to use a Signal

Reply:
Or we could make the signals overrides for java?

Reply:
that is an appealing thought but I see two reasons against:
* we can't easily add more signals, since adding a new method to the abstract class could clash with an existing method defined by user code
* makes signals look different than messages (callbacks vs message processing)

Reply:
There were plenty of missing calls to postStop, I have added them.

Reply:
:+1: